src/app/pages/blog/blog.js

```diff
@@ -3,61 +3,161 @@ import settings from 'settings';
 import {on} from '~/helpers/controller/decorators';
 import routerBus from '~/helpers/router-bus';
 import analytics from '~/helpers/analytics';
-import Article from './article/article';
-import {description as template} from './blog.html';
 import css from './blog.css';
+import SearchBar from './search-bar/search-bar';
+import SearchResults from './search-results/search-results';
+import UpdateBox from './update-box/update-box';
+import FeaturedArticle from './pinned-article/pinned-article';
+import MoreStories from './more-stories/more-stories';
+import {blurbModel} from './article-summary/article-summary';
+import Article from './article/article';
+import DisqusForm from './disqus-form/disqus-form';
 
+function slugsSortedByArticleDate(articles) {
+    return Object.keys(articles)
+        .sort((a, b) => {
+            const articleA = articles[a];
+            const articleB = articles[b];
+
+            return articleA.date < articleB.date ? 1 : -1;
+        });
+}
+
+const path = '/blog';
 const spec = {
-    template,
     css,
     view: {
         classes: ['blog', 'page'],
         tag: 'main'
     },
-    regions: {
-        articles: '.articles .container',
-        pinned: '.pinned',
-        articlePage: '.article.page'
-    },
     slug: 'news',
-    model: {
-        rssUrl: `${settings.apiOrigin}/blog-feed/rss/`
-    }
+    timersRunning: [],
+    articleSlug: null
 };
 const BaseClass = componentType(spec, canonicalLinkMixin, loaderMixin);
 
 export default class Blog extends BaseClass {
 
     init() {
         super.init();
-        this.timersRunning = [];
-        this.previousSlug = null;
+        this.hpcListener = this.handlePathChange.bind(this);
+        window.addEventListener('navigate', this.hpcListener);
+    }
+
+    get featuredArticleOptions() {
+        if (!this.pageData) {
+            return {};
+        }
+        const articleSlug = Reflect.ownKeys(this.pageData.articles)
+            .find((k) => this.pageData.articles[k].pin_to_top);
+        const pinnedData = this.pageData.articles[articleSlug];
+
+        return {
+            model: blurbModel(articleSlug, pinnedData)
+        };
+    }
 
-        // eslint-disable-next-line complexity
-        this.handlePathChange = () => {
-            if (history && history.state && history.state.model) {
-                Object.assign(this.model, history.state.model);
+    moreStoriesOptions(test) {
+        if (!this.pageData) {
+            return {};
+        }
+        const articles = this.pageData.articles;
+        const slugs = Reflect.ownKeys(articles)
+            .filter((k) => test(k, articles[k]))
+            .sort((a, b) => articles[a].date < articles[b].date ? 1 : -1);
+
+        return slugs.map((s) => blurbModel(s, this.pageData.articles[s]));
+    }
+
+    buildDefaultPage() {
+        const fa = new FeaturedArticle(this.featuredArticleOptions);
+        // const sb = new SearchBar({
```

Comment:
Do we need the commented out part?

Reply:
This is a future feature. I'll add a comment to the comments. :)